### ISSUE TITLE
SA1002: SemicolonsMustBeSpacedCorrectly

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#8]
http://www.stylecop.com/docs/SA1002.html

**Comment by jackwhelpton** [comment]
The implementation of this rule is a little different from the original StyleCop version when it comes to the sections of a "for" loop. In the previous version, you would be nudged towards writing an "infinite" for loop as:

````csharp
for (;;)
{
}
````

Other variants would trigger a violation of SA1002. By contrast, the only way to adhere to the rule now appears to be:

````csharp
for (; ;)
{
}
````

which raised a violation of SA1002 in the previous version. Is this deliberate? The same difference is apparent if you add just an initializer to the loop.

I'm posting several queries of this style as I migrate an existing solution to use the StyleCop Analyzers; is it generally intended that the rules that existed in the previous version should preserve their behaviour?

**Comment by sharwell** [comment]
@jackwhelpton It's normal for replies to "get lost" when they are added to issues that are already closed. You should open a new issue for items like this to make sure at a minimum that they get documented.

:memo: Your comment is indirectly related to #633.

**Comment by jackwhelpton** [comment]
Raised https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/1489: cheers.
